server/host_controls_api.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"net/http"
+)
+
+func (p *Plugin) handleMakeHost(w http.ResponseWriter, r *http.Request) {
+	var res httpResponse
+	defer p.httpAudit("handleMakeHost", &res, w, r)
+
+	userID := r.Header.Get("Mattermost-User-Id")
+	callID := mux.Vars(r)["call_id"]
+
+	var params struct {
+		NewHostID string `json:"new_host_id"`
+	}
```

Comment:
Fine for now but eventually should go into `public` so that it can be used by potentially automated or custom clients as well.